ARMA remove FourierOrder

PaulTalbot-INL opened the pull request:
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #909

##### What are the significant changes in functionality due to this change request?
Removes the FourierOrder node in the ARMA in favor of using the Fourier node.

A conversion script is included, and an email to the user list will be required.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

PaulTalbot-INL commented:
The tests have all passed, this is ready for review (and a predecessor to the cluster  ROM MR).

wangcj05 commented on framework/SupervisedLearning/ARMA.py:
```diff
@@ -752,149 +731,61 @@ def _trainCDF(self,data):
-      fourierEvaluations = dict((period,vals[self.zeroFilterMask]) for period,vals in fourierSeriesOriginal.items())
+      # TODO is this right?
+      fourierSignals = fourierSignalsFull[self.zeroFilterMask, :]
+      # OLD fourierSignals = dict((period, vals[self.zeroFilterMask]) for period, vals in fourierSeriesOriginal.items())
```
It seems to me you need to fix the "TODO" here.

PaulTalbot-INL commented:
Addressed comments (removed commented lines, and added in conversion script)

PaulTalbot-INL commented:
User email:
```
RAVEN Users:

As of MR #910, we've simplified the input of the ARMA once again. This time, to decrease the chance of misunderstanding, we've removed the "FourierOrder" node, which was an option to subidivide the Fourier periods selected. Now, each Fourier period desired should be listed explicitly in the "Fourier" node.

To ease transition, a conversion script has been placed in raven/scripts/conversionScripts/ARMA_remove_fourier_orders.py

Thank you!

- RAVEN developers
```

wangcj05 commented:
@PaulTalbot-INL Windows test failed.

PaulTalbot-INL commented:
Tests are passed with the new j4 l4 setup.

wangcj05 commented:
Manual got updated, tests passed, and conversion script is provided. 
PR Checklist is satisfied, and this PR can be merged.

wangcj05 commented:
@PaulTalbot-INL You can send out the email now.
